packages/core-sdk/src/types/resources/license.ts

```diff
@@ -4,92 +4,88 @@ import { TxOptions } from "../options";
-  /*Commercializers that are allowed to commercially exploit the work. If zero address, then no restrictions is enforced.*/
+  /** Commercializers that are allowed to commercially exploit the work. If zero address, then no restrictions is enforced.*/
   commercializerChecker: Address;
-  /*The data to be passed to the commercializer checker contract.*/
+  /** The data to be passed to the commercializer checker contract.*/
   commercializerCheckerData: Address;
-  /*Percentage of revenue that must be shared with the licensor.*/
-  commercialRevShare: number;
-  /*The maximum revenue that can be generated from the commercial use of the work.*/
-  commercialRevCeiling: bigint;
-  /*Indicates whether the licensee can create derivatives of his work or not.*/
+  /** The maximum revenue that can be generated from the commercial use of the work.*/
+  commercialRevCeiling: bigint | string | number;
+  /** Indicates whether the licensee can create derivatives of his work or not.*/
   derivativesAllowed: boolean;
-  /*Indicates whether attribution is required for derivatives of the work or not.*/
+  /** Indicates whether attribution is required for derivatives of the work or not.*/
   derivativesAttribution: boolean;
-  /*Indicates whether the licensor must approve derivatives of the work before they can be linked to the licensor IP ID or not.*/
+  /** Indicates whether the licensor must approve derivatives of the work before they can be linked to the licensor IP ID or not.*/
   derivativesApproval: boolean;
-  /*Indicates whether the licensee must license derivatives of the work under the same terms or not.*/
+  /** Indicates whether the licensee must license derivatives of the work under the same terms or not.*/
   derivativesReciprocal: boolean;
-  /*The maximum revenue that can be generated from the derivative use of the work.*/
-  derivativeRevCeiling: bigint;
-  /*The ERC20 token to be used to pay the minting fee. the token must be registered in story protocol.*/
-  currency: Address;
-  /*The URI of the license terms, which can be used to fetch the offchain license terms.*/
-  uri: string;
-};
-export type RegisterPILTermsRequest = Omit<
-  LicenseTerms,
-  "defaultMintingFee" | "expiration" | "commercialRevCeiling" | "derivativeRevCeiling"
-> & {
-  defaultMintingFee: bigint | string | number;
-  expiration: bigint | string | number;
-  commercialRevCeiling: bigint | string | number;
+  /** The maximum revenue that can be generated from the derivative use of the work.*/
   derivativeRevCeiling: bigint | string | number;
+  /** The URI of the license terms, which can be used to fetch the offchain license terms.*/
+  uri: string;
+}
+/**
+ * The request object that contains all data needed to register a license term.
+ */
+export type RegisterPILTermsRequest<PILType extends PIL_TYPE> = {
+  /**  The license terms to be registered. */
+  terms:
+    | (PILType extends PIL_TYPE.COMMERCIAL_USE ? CommercialLicenseTerms : never)
```

Comment:
I suggest we simplify this usage. For `terms` here in the request, the type should be `PILTermsInput`, which is the same as `RegisterPILTermsRequest` from before. `PilFlavor.xxx` returns `PILTerms`, which should be compatible with `PILTermsInput` as `PILTermsInput` is just `PILTerms` but with some attributes that is now required.
I don't need to allow the user to provide an enum to use a preset, but just need to do `PILFlavor.commercialUse({...})`